File: java/Practica1/src/main/java/deepspace/TestP1.java
/**
 *
 * @author Alejandro Villanueva Prados
 */
package deepspace;

public class TestP1 {
    public static void main(String[] args) {
        // Probamos la clase Loot, creando una instancia suya
        System.out.println("Prueba Loot");
        Loot botin = new Loot(1,2,3,4,5);
        // Veamos si funcionan los consultores
        System.out.println("Suministros: " + botin.getNSupplies());
        System.out.println("Armas: " + botin.getNWeapons());
        System.out.println("Escudos: " + botin.getNShields());
        System.out.println("Hangares: " + botin.getNHangars());
        System.out.println("Medallas: " + botin.getNMedals());
        
        System.out.println("===========================================");
        // Probamos la clase SuppliesPackage, creando una instancia suya
        System.out.println("Prueba SuppliesPackage");
        SuppliesPackage supppkg = new SuppliesPackage(0.5f, 1.5f, 2.2f);
        // Probamos el constructor de copia
        SuppliesPackage otro = new SuppliesPackage(supppkg);
        System.out.println(supppkg.toString() + " == " + otro.toString());
        // Probamos los consultores
        System.out.println("Munición: " + supppkg.getAmmoPower());
        System.out.println("Combustible: " + supppkg.getFuelUnits());
        System.out.println("Escudo: " + supppkg.getShieldPower());
        
        System.out.println("===========================================");
        // Probamos la clase ShieldBooster, creando una instancia suya
        System.out.println("Prueba ShieldBooster");
        ShieldBooster shield = new ShieldBooster("Escudo1", 1.55f, 1);
        // Probamos el constructor de copia.
        ShieldBooster otroshield = new ShieldBooster(shield);
        // Probamos los consultores
        System.out.println(shield.toString() + " == " + otroshield.toString());
        System.out.println("Boost: " + shield.getBoost());
        System.out.println("Usos: " + shield.getUses());
        // Probamos useIt.
        System.out.println("Usando... el boost es: " + shield.useIt());
        System.out.println("Usando de nuevo, ahora: " + shield.useIt());
        
        System.out.println("===========================================");
        // Probamos la clase Weapon, creando una instancia suya
        System.out.println("Prueba Weapon");
        Weapon wp = new Weapon("Ion Cannon", WeaponType.LASER, 1);
        // Probamos el constructor de copia
        Weapon copy = new Weapon(wp);
        System.out.println(wp.toString() + " == " + copy.toString());
        // Probamos los consultores
        System.out.println("Tipo: " + wp.getType());
        System.out.println("Usos: " + wp.getUses());
        // Probamos el método power()
        System.out.println("Power: " + wp.power());
        // Ahora probamos useIt()
        System.out.println("Usamos el arma, potencia: " + wp.useIt());
        System.out.println("De nuevo, ahora: " + wp.useIt());
        
        System.out.println("===========================================");
        // Probamos la clase Dice, creando una instancia suya
        System.out.println("Prueba Dice");
        Dice dado = new Dice();
        
        int hangarcount = 0;
        int weaponcount = 0;
        int shieldcount = 0;
        int shotcount = 0;
        int movecount = 0;
        for (int i = 0; i < 100; ++i) {
            if (dado.initWithNHangars() == 0)
                hangarcount++;
            if (dado.initWithNWeapons() == 1)
                weaponcount++;
            if (dado.initWithNShields() == 0)
                shieldcount++;
            if (dado.firstShot() == GameCharacter.SPACESTATION)
                shotcount++;
            if (dado.SpaceStationMoves(0.5) == true)
                movecount++;
        }
        
        System.out.println("NHANGARPROB: " + hangarcount);
        System.out.println("NWEAPONPROB: " + weaponcount);
        System.out.println("NSHIELDPROB: " + shieldcount);
        System.out.println("NFIRSTSHOTPROB: " + shotcount);
        System.out.println("NMOVEPROB: " + movecount);
        
    }
}

File: ruby/test_p1.rb
#encoding: UTF-8

require_relative "combat_result.rb"
require_relative "game_character.rb"
require_relative "shot_result.rb"
require_relative "weapon_type.rb"
require_relative "loot.rb"
require_relative "supplies_package.rb"
require_relative "shield_booster.rb"
require_relative "weapon.rb"
require_relative "dice.rb" 

module Deepspace
  class TestP1
    def initialize
    end

    public
    def main
      puts "Prueba Loot"
      # Probamos que la clase Loot funciona, creando una instancia suya
      botin = Loot.new(1,2,3,4,5)
      puts botin.inspect
      # Veamos que funcionan los consultores
      puts "Suministros: " + botin.getNSupplies.to_s
      puts "Armas: " + botin.getNWeapons.to_s
      puts "Escudos: " + botin.getNShields.to_s
      puts "Hangares: " + botin.getNHangars.to_s
      puts "Medallas: " + botin.getNMedals.to_s

      puts "=================================================="
      puts "Prueba SuppliesPackage"
      # Probamos que la clase SuppliesPackage funciona, creando una instancia suya
      suppkg = SuppliesPackage.new(0.5, 1.5, 2.2)
      puts suppkg.inspect
      # Probamos el constructor de copia
      otro = SuppliesPackage.newCopy(suppkg)
      puts suppkg.inspect + " == " + otro.inspect
      # Probamos los consultores
      puts "Munición: " + suppkg.getAmmoPower.to_s
      puts "Combustible: " + suppkg.getFuelUnits.to_s
      puts "Escudo: " + suppkg.getShieldPower.to_s

      puts "=================================================="
      puts "Prueba ShieldBooster"
      # Probamos la clase ShieldBooster, creando una instancia suya
      shield = ShieldBooster.new("Escudo1", 1.55, 1)
      puts shield.inspect
      # Probamos el constructor de copia
      copia = ShieldBooster.newCopy(shield)
      puts shield.inspect + " == " + copia.inspect
      # Probamos los consultores
      puts "Boost: " + shield.getBoost.to_s
      puts "Usos: " + shield.getUses.to_s
      # Probamos el método useIt
      puts "Usando... el boost es " + shield.useIt.to_s
      puts "Usando de nuevo, ahora: " + shield.useIt.to_s

      puts "=================================================="
      puts "Prueba Weapon"
      # Probamos la clase Weapon
      w = Weapon.new("Ion Cannon", WeaponType::LASER, 1)
      puts w.inspect
      # Probamos el constructor de copia
      wcopy = Weapon.newCopy(w)
      puts w.inspect + " == " + wcopy.inspect
      # Probamos los consultores
      puts "Tipo: " + w.getType.to_s
      puts "Usos: " + w.getUses.to_s
      # Probamos el método power()
      puts "Potencia: " + w.power().to_s
      # Probamos useIt()
      puts "Usando un disparo: " + w.useIt.to_s
      puts "Otro: " + w.useIt.to_s

      puts "=================================================="
      puts "Prueba Dice"
      # Probamos la clase Dice
      dado = Dice.new
      # Prueba de los métodos:
      hangarcount = 0
      weaponcount = 0
      shieldcount = 0
      sscount = 0
      movecount = 0
      for i in 0..100
        if dado.initWithNHangars == 0
          hangarcount += 1
        end
        if dado.initWithNWeapons == 1 
          weaponcount += 1
        end
        if dado.initWithNShields == 0
          shieldcount += 1
        end
        if dado.firstShot == GameCharacter::SPACESTATION
          sscount += 1
        end
        if dado.spaceStationMoves(0.5) == true
          movecount += 1
        end
      end

      puts "NHANGARPROB: " + hangarcount.to_s
      puts "NWEAPONPROB: " + weaponcount.to_s
      puts "NSHIELDPROB: " + shieldcount.to_s
      puts "NFIRSTSHOTPROB: " + sscount.to_s
      puts "NMOVEPROB: " + movecount.to_s
    end
  end
  t1 = TestP1.new
  t1.main
end
